src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -416,25 +416,21 @@ public void EnsureSolutionIsLoaded()
             });
         }
 
-        public string SolutionDirectory
+        public string SolutionDirectory => NuGetUIThreadHelper.JoinableTaskFactory.Run(GetSolutionDirectoryAsync);
```

Comment:
Where is this used now that there exists `GetSolutionDirectoryAsync` ? Can it be marked as deprecated?

Reply:
There's way too many usages to mark is deprecated for now. 

Most of them are warranted (ie, sync codepaths that can't be used otherwise). 

I do agree with the general sentiment, but I'm not sure we can move it right now.

Reply:
https://github.com/NuGet/NuGet.Client/pull/4238#pullrequestreview-745387230